### PR TITLE
Persist data

### PR DESCRIPTION
**Pull request by sjwoodman**
This branch persists the data sent to it from a client outside Openshift. This is meant to simulate the real world scenario where clients outside an organisation push data into a db via a REST API. Then the data is processed as a stream by the organisation.

I followed one of the lab sessions to create a wildfly swarm app that stores data in JPA. There may be better ways of creating the containers as the current method is quite fragile (deploymentconfig.yml).

https://github.com/redhat-microservices/lab_swarm-openshift

### CODE REVIEW

File: persistent-producer/src/main/java/com/redhat/demo/clnr/rest/PersistentProducerEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    public Reading getReading(@PathParam("id") Long id) {
+
+        return em.find(Reading.class, id);
+    }
+
+    /**
+     * Persist a reading to the DB and send a kafka message
+     * @param r Reading to be stored
+     */
+    private void persistReading(Reading r){
```
Comment (matzew) [inline persistent-producer/src/main/java/com/redhat/demo/clnr/rest/PersistentProducerEndpoint.java]: Not sure I'd do it that way.  IMO the method should not do two things:
* store data in DB
* store data in Kafka

**Comment by matzew** [comment]
I'd implement the "store data in MySQL" differently

I'd have one single API endpoint, that read the data / parses the CVS _and_ hammers the data into Kafka.

A different consumer could be than just read from the topic and pump the data into MySQL

**Comment by sjwoodman** [comment]
Good point - I'll refactor the app to reflect this architecture

**Comment by sjwoodman** [comment]
Closed and will open another with comments addressed

**Pull request closed without merging**